File: LeetCode/Medium/0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>> res;
        unordered_map<string, vector<string>> um;

        for(auto& str: strs){
            string token = str;
            sort(token.begin(), token.end());

            um[token].push_back(str);
        }

        for(auto& p: um){
            res.push_back(p.second);
        }

        return res;
    }
};

File: LeetCode/Medium/0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> m = new HashMap<>();

        for(String str: strs){
            char[] c_str = str.toCharArray();
            Arrays.sort(c_str);
            String key = new String(c_str);

            m.putIfAbsent(key, new ArrayList<String>());
            m.get(key).add(str);
        }

        return new ArrayList<>(m.values());
    }
}